README.md

```diff
@@ -1 +1,167 @@
+
+```js
+callOne({...}, function(error, data1) {
+    callTwo(data1, function(error, data2) {
+        callThree(data2, function(error, data3) {
+        ...
+        })
+    })
+})
+```
+
+Remember, this is asynchronous world so we cannot type `callOne` then `callTwo` on a new line. We need to use callback and with callbacks (the way they are written in the example above) comes indentation.
+
+Now add on top of that a few callbacks to perform something else essential like opening a database connection and starting a server. Look at this:
+
+
+```js
+fs.readdir(source, function (err, files) {
+  if (err) {
+    console.log('Error finding files: ' + err)
+  } else {
+    files.forEach(function (filename, fileIndex) {
+      console.log(filename)
+      gm(source + filename).size(function (err, values) {
+        if (err) {
+          console.log('Error identifying file size: ' + err)
+        } else {
+          console.log(filename + ' : ' + values)
+          aspect = (values.width / values.height)
+          widths.forEach(function (width, widthIndex) {
+            height = Math.round(width / aspect)
+            console.log('resizing ' + filename + 'to ' + height + 'x' + height)
+            this.resize(width, height).write(dest + 'w' + width + '_' + filename, function(err) {
+              if (err) console.log('Error writing file: ' + err)
+            })
+          }.bind(this))
+        }
+      })
+    })
+  }
+}
+```
+
+You might end up with something called the nested approach, pyramid of doom or callback hell. This code is very hard to read and maintain. People who have to work on this code after you might start to hate you!
+
+The `readdir` example is from <http://callbackhell.com> (yes, there is even a website for Callback Hell!).
+
+This lesson will cover the ways to combat callback hell.
+
+## Objectives
+
+1. Explain that a callback hell script is a complication of the callback async pattern
+1. Identify a callback hell script
+1. Implement possible ways to combat callback hell with named functions
+1. Implement possible ways to combat callback hell with modules
+
+
+## Using A Named Function to Mitigate Callback Hell
+
+The easiest way to reduce the level of indentation and thus reduce callback hell is to use named functions. Named functions are functions with names or variables which values are functions.
+
+```js
+function processResponse1(error, data1) {
+  ...
+}
+
+callOne({...}, processResponse1)
+```
+
+or
+
+```js
+var processResponse1 = function(error, data1) {
+  ...
+}
+
+callOne({...}, processResponse1)
+```
+
+The difference is that the named function will be hoisted, meaning its definition can be anywhere in the file while the variable function's definition must precede the usage of it. Another difference between named functions and variables is that named function will have names in the stack trace (the error message for debugging) so they are better for trouble shooting and debugging..
```

Comment:
I'm also not sure these two examples are necessary. This is one case where being opinionated is great, and it sounds like (and judging by below), we can pretty confidently proceed with named functions.